scripts/flow/v11/flowListener.js

```diff
@@ -0,0 +1,115 @@
+			length: Math.max(
+				state.data.acc_diff?.targets?.length || 0,
+				state.data.hit_results?.length || 0,
+			),
+		},
+		(_, i) => ({
+			target: state.data.acc_diff?.targets?.[i],
+			hit_result: state.data.hit_results?.[i],
+		}),
+	);
+
+	return new FlowInfo({
+		sourceToken: getTokenByIdOrActorId(state.actor.token?.id || state.actor?.id),
+		actionIdentifier: state.item?.system?.lid || fallbackActionIdentifier,
+		targetTokens: zippedTargetInfo
+			.map(({target}) => target.target)
+			.filter(Boolean),
+		targetsMissed: new Set(
+			zippedTargetInfo
+				.filter(({hit_result}) => !hit_result?.hit)
+				.map(({target}) => target?.target?.id)
+				.filter(Boolean)
+		),
+	});
+}
+
+/**
+ * @param {string} flowName
+ * @param {?((string|Function))} fallbackActionIdentifier
+ */
+const _bindFlowHook = (
+	{
+		flowName,
+		fallbackActionIdentifier = null,
+	}
+) => {
+	Hooks.on(`lancer.postFlow.${flowName}`, (flow, isSuccess) => {
+		if (!isSuccess) return;
+
+		if (fallbackActionIdentifier != null && fallbackActionIdentifier instanceof Function) {
+			fallbackActionIdentifier = fallbackActionIdentifier(flow);
+		}
+
+		const flowInfo = _getFlowInfo(
+			flow.state,
+			{
+				fallbackActionIdentifier
+			},
+		);
+		if (flowInfo == null) return;
+
+		processFlowInfo(flowInfo);
+	});
+};
+
+const _onReady = () => {
+	if (!foundry.utils.isNewerVersion(game.version, "11")) return;
+
+	// Weapon attacks
+	_bindFlowHook({flowName: "WeaponAttackFlow"});
+	// Basic attacks
+	_bindFlowHook({
+		flowName: "BasicAttackFlow",
+		fallbackActionIdentifier: flow => flow.state.data.attack_type === "Melee"
+			? "lwfx_default_melee"
+			: "lwfx_default_ranged",
+	});
+
+	// Tech attacks and invades
+	_bindFlowHook({flowName: "TechAttackFlow", fallbackActionIdentifier: "default_tech_attack"});
+
+	// Stabilize
+	_bindFlowHook({flowName: "StabilizeFlow", fallbackActionIdentifier: "lwfx_stabilize"});
+	// Full repair
+	_bindFlowHook({flowName: "FullRepairFlow", fallbackActionIdentifier: "lwfx_stabilize"});
+
+	// Core power
+	_bindFlowHook({flowName: "CoreActiveFlow"});
+
+	// Other activations, which use an action
+	// E.g. "Pattern-A Smoke Charges" -> "Use Quick"
+	_bindFlowHook({flowName: "ActivationFlow"});
+
+	// Other activations, which do not use an action
+	// E.g. "Rapid Burst Jump Jet System" -> "Use"
+	_bindFlowHook({flowName: "SystemFlow"});
+
+	// Overcharge
+	_bindFlowHook({flowName: "OverchargeFlow", fallbackActionIdentifier: "lwfx_overcharge"});
+
+	// Structure
+	_bindFlowHook({flowName: "StructureFlow", fallbackActionIdentifier: "lwfx_structure"});
+	// Structure side effects (equipment destruction)
+	_bindFlowHook({flowName: "SecondaryStructureFlow", fallbackActionIdentifier: "lwfx_structure_secondary"});
+
+	// Stress
+	_bindFlowHook({flowName: "OverheatFlow", fallbackActionIdentifier: "lwfx_overheat"});
+
+	// Cascades (structure/stress side effect)
+	_bindFlowHook({flowName: "CascadeFlow", fallbackActionIdentifier: "lwfx_cascade"});
```

Comment:
I have added all the flows which I think could make good use of Weapon FX in future, even if they do not have any animation currently.

New custom IDs which could be given FX in future are (fyi @Z3nner !):

```
lwfx_overcharge
lwfx_structure
lwfx_structure_secondary
lwfx_overheat
lwfx_cascade
```

The flows I did _not_ add hooks for, as I was unsure how they would be represented visually, are:

```
BondPowerFlow -- May be good to add?
NPCRechargeFlow -- May be good to add?
TalentFlow -- May be good to add?

StatRollFlow -- Single stat roll hard to visualize?

ActionTrackFlow -- UI flow, so not relevant?
SimpleHTMLFlow -- N/A
SimpleTextFlow -- N/A
```

Reply:
Wow, so I could make an overcharge macro, use that reference you just gave, and link the two together just like I have been with weapons?

Reply:
Yes, exactly! If I add it as `"lwfx_overcharge": "Stabilize"` and re-use that animation as an example, here is how it looks:

[overcharge.webm](https://github.com/Z3nner/lancer-weapon-fx/assets/9784519/4d0e916c-9b66-49c4-923e-f5b3844c1b8d)

Reply:
Well I'm excited now.  Definitely looking forward to running with that to see what other actions I can break into.

I unfortunately don't know enough about these sort of things to contribute further to the conversation though, so I'll defer further action to @dodgepong and the rest of you.